spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureCloudConfigAutoConfiguration.java

```diff
@@ -17,7 +17,8 @@
 
 @Configuration
 @ConditionalOnClass(RefreshEndpoint.class)
-@ConditionalOnProperty(name = AzureCloudConfigProperties.CONFIG_PREFIX + ".watch.enabled", matchIfMissing = true)
+@ConditionalOnProperty(prefix = AzureCloudConfigProperties.CONFIG_PREFIX, name = {"enabled", "watch.enabled"},
```

Comment:
I suppose you still need to trigger this auto config when watch is disabled.

Reply:
Currently, it only contains auto configuration for watch related beans, so when watch is disabled, no bean will be created.

Reply:
Got it. Then class name might be better to contain `Watch` if this is the sole purpose.

Reply:
OK, moved the watch related configuration into separate class, make the configuration more clear.